x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java

```diff
@@ -94,12 +94,11 @@ public class EnterpriseSearch extends Plugin implements ActionPlugin, SystemInde
 
     private final boolean enabled;
 
+    private final boolean queryRulesEnabled;
+
     public EnterpriseSearch(Settings settings) {
         this.enabled = XPackSettings.ENTERPRISE_SEARCH_ENABLED.get(settings);
-    }
-
-    protected XPackLicenseState getLicenseState() {
```

Comment:
I _think_ that with the recent changes from #96570 you will need to merge main first, because removing this method will cause build failures.

Reply:
Re-adding this back after a rebase 👍